gateway/src/shard/processor/impl.rs

```diff
@@ -201,7 +201,7 @@ impl ShardProcessor {
             large_threshold: 250,
             properties: self.properties.clone(),
             shard: Some(self.config.shard()),
-            presence: None,
+            presence: self.config.presence(),
```

Comment:
How come you opted for swapping the cloning around? Imo it would make more sense for `pub fn presence()` to return a borrowed `UpdateStatusInfo` and then in here to just `.clone()` that.

Reply:
because the shard wants a UpdateStatusInfo instead of a borrowed one and it would be far messier to clone it here as it requires matching the thing first

Reply:
You can use `Option::cloned`: <https://doc.rust-lang.org/std/option/enum.Option.html#method.cloned>

Reply:
huh, didn't know that's a thing, i'll swap it out